pynetbox/core/endpoint.py

```diff
@@ -20,6 +20,37 @@
 RESERVED_KWARGS = ()
 
 
+class CachedRecordRegistry:
```

Comment:
Maybe? Split this and the one in core/app.py into two separate PR's?  Although both are caching - the seem like functionally separate functionality that can be reviewed separately (and potentially backed out separately if any issues).